File: app/src/main/java/com/ragdroid/dahaka/activity/home/profile/ProfilePresenter.java
package com.ragdroid.dahaka.activity.home.profile;

import com.ragdroid.dahaka.api.entity.Pokemon;
import com.ragdroid.dahaka.app.UserManager;
import com.ragdroid.dahaka.mvp.BasePresenterImpl;

import java.net.MalformedURLException;
import java.net.URL;

import javax.inject.Inject;

/**
 * Created by garimajain on 17/08/17.
 */

public class ProfilePresenter extends BasePresenterImpl<ProfileContract.View> implements ProfileContract.Presenter {

    private final Pokemon pokemon;

    @Inject
    public ProfilePresenter(Pokemon pokemon) {
        this.pokemon = pokemon;
    }

    @Override
    public void onViewAdded(ProfileContract.View view) {
        super.onViewAdded(view);
        URL url = null;
        try {
            url = new URL(pokemon.getSprites().getFrontDefault());
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        ProfileModel model = ProfileModel.newBuilder()
                .experience(pokemon.getBaseExperience())
                .height(pokemon.getHeight())
                .imageUrl(url != null ? url.toString() : "")
                .weight(pokemon.getWeight())
                .name(pokemon.getName())
                .build();
        getView().showModel(model);

    }
}

File: app/src/main/java/com/ragdroid/dahaka/activity/home/profile/ProfilePresenter.kt
package com.ragdroid.dahaka.activity.home.profile

import com.ragdroid.dahaka.api.entity.Pokemon
import com.ragdroid.dahaka.app.UserManager
import com.ragdroid.dahaka.mvp.BasePresenterImpl

import java.net.MalformedURLException
import java.net.URL

import javax.inject.Inject

/**
 * Created by garimajain on 17/08/17.
 */

class ProfilePresenter @Inject
constructor(private val pokemon: Pokemon) : BasePresenterImpl<ProfileContract.View>(), ProfileContract.Presenter {

    override fun onViewAdded(view: ProfileContract.View) {
        super.onViewAdded(view)
        var url: URL? = null
        try {
            url = URL(pokemon.sprites?.frontDefault)
        } catch (e: MalformedURLException) {
            e.printStackTrace()
        }

        val model = ProfileModel(
                if (url != null) url.toString() else "",
                pokemon.name,
                pokemon.weight ?: 0,
                pokemon.height ?: 0,
                pokemon.baseExperience ?: 0)

        view.showModel(model)

    }
}
